### PR TITLE
Docs: Add details about native CLI usage

### PR DESCRIPTION
**Pull request by oraNod**

### CODE REVIEW

File: documentation/asciidoc/topics/proc_connecting_cli.adoc [hunk 1]
```diff
@@ -27,12 +34,25 @@ $ {kubectl_exec}
...
 [source,bash,options="nowrap",subs=attributes+]
 ----
 $ bin/cli.sh -c https://$SERVICE_HOSTNAME:$PORT --trustall
 ----
+ifdef::downstream[]
++
+.Using the native CLI
+[source,bash,options="nowrap",subs=attributes+]
+----
+$ ./datagrid-cli -c https://$SERVICE_HOSTNAME:$PORT
```
Comment (ryanemerson) [inline documentation/asciidoc/topics/proc_connecting_cli.adoc:53]: I don't think we can merge this until https://github.com/infinispan/infinispan-quarkus/pull/61 is merged and available downstrream. We may have missed the boat for GA \cc @pruivo

**Comment by Crumby** [comment]
I wonder if it'd be better to drop the parts talking about getting CLI as part of the server distribution and keep native only in the context of operator.

**Comment by ryanemerson** [comment]
> I wonder if it'd be better to drop the parts talking about getting CLI as part of the server distribution and keep native only in the context of operator.

It's tricky for a few reasons:
- Upstream we also have the `infinispan/cli` image
- The native cli is only available for linux

File: documentation/asciidoc/topics/proc_connecting_cli.adoc (hunk 1 above)
Comment (pruivo) [reply documentation/asciidoc/topics/proc_connecting_cli.adoc]: yes, the commit of that PR is applied in the internal repository.

File: documentation/asciidoc/topics/proc_connecting_cli.adoc [hunk 2]
```diff
@@ -6,15 +6,22 @@ Use the command line interface (CLI) to connect to your {brandname} cluster and
...
+The CLI is available as part of the server distribution.
+Alternatively, you can use the CLI container image available from link:https://infinispan.org/download/stable/cli-image.html[infinispan.org/download].
 endif::community[]
 
+//Downstream content
+ifdef::downstream[]
+The {brandname} CLI is available with the server distribution or as a native executable.
+
+Follow the instructions in _Getting Started with {brandname} Server_ for information on downloading and installing the CLI as part of the server distribution.
+For the native CLI, you should follow the installation instructions in the _README_ file that is included with the distribution.
```
Comment (pruivo) [inline documentation/asciidoc/topics/proc_connecting_cli.adoc:22]: what README file?

File: documentation/asciidoc/topics/proc_connecting_cli.adoc [hunk 3]
```diff
@@ -27,12 +34,25 @@ $ {kubectl_exec}
...
 [source,bash,options="nowrap",subs=attributes+]
 ----
 $ bin/cli.sh -c https://$SERVICE_HOSTNAME:$PORT --trustall
 ----
+ifdef::downstream[]
++
+.Using the native CLI
+[source,bash,options="nowrap",subs=attributes+]
+----
+$ ./redhat-datagrid-cli -c https://$SERVICE_HOSTNAME:$PORT
```
Comment (pruivo) [inline documentation/asciidoc/topics/proc_connecting_cli.adoc:53]: I guess you can add `--trustall` so the commands (native vs non-native) are identical

File: documentation/asciidoc/topics/proc_connecting_cli.adoc (hunk 2 above)
Comment (oraNod) [reply documentation/asciidoc/topics/proc_connecting_cli.adoc:22]: README.adoc - that's in the zip that comes with CR2

Comment (pruivo) [reply documentation/asciidoc/topics/proc_connecting_cli.adoc:22]: I'm not sure if it is clear. the word "distribution" is always after "server" and users may think the native executable is in the "server distribution"

Comment (oraNod) [reply documentation/asciidoc/topics/proc_connecting_cli.adoc:22]: gotcha. I'll reword. thanks!

**Review by ryanemerson — APPROVED** [review]

**Review by pruivo — APPROVED** [review]

**Comment by ryanemerson** [comment]
Thanks @oraNod

**Pull request merged**